kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/render/AbstractFormRenderer.java

```diff
@@ -776,22 +815,33 @@ protected String appendExtractionExpression(String type, String name, String id,
             jsonTemplate.append("getMultipleInputData('")
                         .append(id)
                         .append("')");
-        } else if (type.equals("datetime-local")) {
-            if (isShowTime) {
-                jsonTemplate.append("document.getElementById('")
-                        .append(id)
-                        .append("')")
-                        .append(getExtractionValue(jsType));
+        } else if (isDate(type)) {
+            if (type.equals("datetime-local")) {
+                if (isShowTime) {
+                    jsonTemplate.append("getDate('")
+                            .append(id)
+                            .append("',")
+                            .append(FUNCTION_MAPPING.get(formFieldType))
+                            .append(") ");
+                } else {
+                    jsonTemplate.append("getDateWithoutTime('")
+                            .append(id)
+                            .append("',")
+                            .append(FUNCTION_MAPPING.get(formFieldType))
+                            .append(") ");
+                }
```

Comment:
We can save the duplicated `if else` branches by turning `if` part into following, and keeping `else` as you wrote below.
```suggestion
            if (type.equals("datetime-local") && !showTime) {
                jsonTemplate.append("getDateWithoutTime('")
                        .append(id)
                        .append("',")
                        .append(FUNCTION_MAPPING.get(formFieldType))
                        .append(") ");
```

Reply:
yes